### PR TITLE
Delete deprecated dockerized-e2e-runner.sh

### PR DESCRIPTION
**Pull request by fejta**
We moved this file to the test-infra repo in https://github.com/kubernetes/test-infra/pull/385

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30937)

### CODE REVIEW

**Comment by ixdy** [comment]
are we 100% sure that none of the builds are using this? (especially non-master branches.)

**Comment by fejta** [comment]
https://github.com/kubernetes/test-infra/search?utf8=%E2%9C%93&q=dockerized-e2e-runner.sh

**Comment by ixdy** [comment]
https://github.com/kubernetes/kubernetes/blob/ff9980e2035210f14a4a9a3f6c756f28c23c883a/docs/devel/development.md#L147 probably needs updating?

File: docs/devel/development.md [hunk 1]
```diff
@@ -144,7 +144,7 @@ unique places you need to update the go version.
 - The image for cross compiling in [build/build-image/cross/](../../build/build-image/cross/). The `VERSION` file and `Dockerfile`.
 - The jenkins test-image in
   [hack/jenkins/test-image/](../../hack/jenkins/test-image/). The `Dockerfile` and `Makefile`.
```
Comment (ixdy) [inline docs/devel/development.md]: probably worth mentioning that the e2e image then also needs to be updated?

Comment (fejta) [inline docs/devel/development.md]: Done

**Comment by ixdy** [comment]
LGTM

**Pull request merged**